x/emissions/module/network_summaries.go

```diff
@@ -115,108 +109,96 @@ func CalcForcastImpliedInferencesAtTime(
 				weightInferenceDotProduct += w_ijk * inferenceByWorker[forecast.ForecastElements[j].Inferer].Value
 				weightSum += w_ijk
 			}
-			I_i[k] = emissions.Inference{
+			forecast := emissions.Inference{
 				Worker: forecast.Forecaster,
 				Value:  weightInferenceDotProduct / weightSum,
 			}
+			I_i[k] = &forecast
 		}
 	}
 
 	// A value of 0 => no inference corresponded to any of the forecasts from a forecaster
 	return I_i, nil
 }
 
-// Gathers inferences and forecasts from the worker update at or just after the given time, and
-// losses from the loss update just before the given time.
-// Then invokes calculation of the forecast-implied inferences using the Inference Synthesis formula from the litepaper.
-func GetAndCalcForcastImpliedInferencesAtBlock(
-	ctx sdk.Context,
-	k keeper.Keeper,
-	topicId TopicId,
-	blockHeight BlockHeight) ([]emissions.Inference, error) {
-	// Get inferences from worker update at or just after the given time
-	inferences, err := k.GetInferencesAtOrAfterBlock(ctx, topicId, blockHeight)
-	if err != nil {
-		fmt.Println("Error getting inferences: ", err)
-		return nil, err
+func MakeMapFromWorkerToTheirWork(inferences []*emissions.Inference) map[string]*emissions.Inference {
+	inferencesByWorker := make(map[string]*emissions.Inference)
+	for _, inference := range inferences {
+		inferencesByWorker[inference.Worker] = inference
 	}
-	// Get forecasts from worker update at or just after the given time
-	forecasts, err := k.GetForecastsAtOrAfterBlock(ctx, topicId, blockHeight)
-	if err != nil {
-		fmt.Println("Error getting forecasts: ", err)
-		return nil, err
+	return inferencesByWorker
+}
+
+type RegretsByWorkerByType struct {
+	InferenceRegrets *map[string]*float64
+	ForecastRegrets  *map[string]*float64
+}
+
```

Comment:
a comment for this function describing it wouldn't be the worst even if the name is pretty clear (or why it's a necessary function or something to give a little context to someone skimming the PR e.g. me)